pkg/reader/ccip.go

```diff
@@ -1166,17 +1187,20 @@ func (r *ccipChainReader) getOnRampDestChainConfig(
 			continue
 		}
 
+		// For chain X, all DestChainConfigs will have same OnRamp address which is the chainX OnRamp
 		chainSel := chainSel
 		eg.Go(func() error {
-			// read onramp dynamic config
 			resp := onRampDestChainConfig{}
 			err := r.contractReaders[chainSel].ExtendedGetLatestValue(
 				ctx,
 				consts.ContractNameOnRamp,
 				consts.MethodNameOnRampGetDestChainConfig,
 				primitives.Unconfirmed,
 				map[string]any{
-					"destChainSelector": chainSel,
+					// Any other chain other than self should work as a chain don't have self as a destination
+					// Here we use destChain as the chainSel is looping over all source chains, so all of them should have
+					// a config for the destChain
```

Comment:
Wait, this comment doesn't make sense to me. You're getting the dest chain config from each onramp, correct? Then we should always use `r.destChain`. I believe this was the fix I made here but elected not to merge: https://github.com/smartcontractkit/chainlink-ccip/pull/236/files#diff-a16bcafc2684d6be248d96e8fdf5fa77ec115f311c0c73d13f3f942ad0edb0c4R1123

Reply:
> You're getting the dest chain config from each onramp

Yes

> Then we should always use r.destChain

All dest configs have same router address (or maybe test address). Why should it be always `r.destChain`?  In this specific line of code it needs to be `r.destChain` but in general any address other than self that has a destChainConfig should work. I'll rephrase it to reduce confusion.